src/v/storage/segment.h

```diff
@@ -327,7 +335,7 @@ class segment {
     offset_tracker _tracker;
     segment_reader_ptr _reader;
     segment_index _idx;
-    bitflags _flags{bitflags::none};
+    bitflags _flags{bitflags::may_have_tombstone_records};
```

Comment:
dislike this construction but unless we flip the meaning of the bit (`does_not_have_tombstone_records`, ugh) it's necessary

Reply:
Moved this marker to the segment index.